### PR TITLE
Namespaced facts fixes

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY

remove 'ansible_' from namespaced facts (redundant)
Shorten warning on facts collisions

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

facts
##### ANSIBLE VERSION

```
2.4
```

### CODE REVIEW

**Comment by alikins** [comment]
If we can pass the c.NAMESPACE_FACTS setting to setup.py, it could learn to collect without the 'ansible_' prefix namespace so the var combing code wouldn't need to.

**Comment by bcoca** [comment]
setup is not only module that does that

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/28040/9/tests

```
{
    "failed": true, 
    "msg": "The conditional check 'ansible_os_family in ['Debian', 'Suse']' failed. The error was: error while evaluating conditional (ansible_os_family in ['Debian', 'Suse']): 'ansible_os_family' is undefined\n\nThe error appears to have been in '/root/ansible/test/integration/targets/apache2_module/tasks/actualtest.yml': line 16, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: install apache via apt\n  ^ here\n"
}
```

**Comment by bcoca** [comment]
*sigh* task executor modified both, but we really dont need that except for 'when_x' results processing, which should be fine with original facts. Removed the code from there.

File: lib/ansible/config/data/config.yml [hunk 1]
```diff
@@ -1257,6 +1248,17 @@ NETWORK_GROUP_MODULES:
...
+  default: False
+  desc:
+    - Facts normally get injected as top level variables, this setting prevents that.
+    - Facts are still available in the `ansible_facts` variable w/o the `ansible_` prefix.
+  env: [{name: ANSIBLE_RESTRICT_FACTS}]
+  ini:
+  - {key: restrict_facts_namespace, section: defaults}
+  value_type: boolean
+  vars: []
+  yaml: {key: defaults.restrict_facts_namespace}
```
Comment (mattclay) [inline lib/ansible/config/data/config.yml:1263]: Shouldn't this have `version_added: "2.4"` also?

**Comment by mattclay** [comment]
CI failure in integration tests: https://app.shippable.com/github/ansible/ansible/runs/28077/25/tests

```
{
    "failed": true, 
    "msg": "The conditional check 'ansible_fips|bool != True' failed. The error was: error while evaluating conditional (ansible_fips|bool != True): 'ansible_fips' is undefined\n\nThe error appears to have been in '/root/ansible/test/integration/targets/copy/tasks/main.yml': line 64, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: verify that the legacy md5sum is correct\n  ^ here\n"
}
```

**Pull request merged**